lib/r2d2/google_payment_token.rb

```diff
@@ -0,0 +1,26 @@
+module R2D2
+  class GooglePaymentToken < PaymentToken
+    def initialize(token_attrs)
+      version = token_attrs['protocolVersion']
+      raise ArgumentError, "unknown protocolVersion #{version}" unless version == 'ECv1'
+
+      signed_message = JSON.parse(token_attrs['signedMessage'])
+      super(signed_message)
+    end
+
+    def decrypt(private_key_pem)
+      message = super
+
+      payment_method = message['paymentMethod']
+      raise ArgumentError, "unknown paymentMethod #{payment_method}" unless payment_method == 'TOKENIZED_CARD'
+
+      message['paymentMethodDetails']
+    end
+
+    class << self
```

Comment:
best to move classes/class methods to the top of the file to avoid them being in a private section (which will have no effect other than confuse people)